chapter_4/ex05_config_file_package/CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+# CMakeLists file for the Chapter 4 Example nr. 05: 
+# Config-file package
+#
+# SPDX-License-Identifier: MIT
+
+project(
+  ch4_ex05_lib
+  VERSION 1.0
+  DESCRIPTION "Chapter 4 Example 05, Config-file package"
+  LANGUAGES CXX)
+
+
+include(GNUInstallDirs) # Defines the ${CMAKE_INSTALL_INCLUDEDIR} variable.
+
+# This is the default install directory for config-file package cmake files
+# It is relative to install prefix.
+set(CFP_INSTALL_DIR cmake CACHE PATH "Installation directory for config-file package cmake files")
```

Comment:
I don't have the receipts, but I asked Robert Schumacher (vcpkg) about what is the recommended way to make this the most flexible and his recommendation was naming such a variable `<project-name>_INSTALL_CMAKEDIR`, which makes it a project specific variable, but also mimics the style of the variables that `GNUInstallDirs` creates. Using a value `"${CMAKE_INSTALL_DATADIR}/<package>"` will also make it easier to create a port for a project when used for vcpkg, see [here](https://github.com/friendlyanon/cmake-init-vcpkg-example/blob/master/ports/cmake-init-header-only/portfile.cmake). Note the lack of any options passed in the configure command.

Reply:
If there is a specific convention, it is better to follow it. I'll rename the variable.

Reply:
It's not really a convention, but merely a suggestion I received from one of the vcpkg devs :)

The value however is really up to you, but it should definitely be something that by default [`find_package` can find](https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure). making it `"${CMAKE_INSTALL_DATADIR}/<package>"` by default will just make writing vcpkg ports easier.